spark3-extensions/src/main/scala/org/apache/iceberg/spark/extensions/IcebergSparkSessionExtensions.scala

```diff
@@ -43,6 +43,7 @@ class IcebergSparkSessionExtensions extends (SparkSessionExtensions => Unit) {
     // TODO: PullupCorrelatedPredicates should handle row-level operations
     extensions.injectOptimizerRule { _ => PullupCorrelatedPredicatesInRowLevelOperations }
     extensions.injectOptimizerRule { _ => RewriteDelete }
+    extensions.injectOptimizerRule { _ => RewriteMergeInto }
```

Comment:
@dilipbiswal, do we have to add MERGE operations to `PullupCorrelatedPredicatesInRowLevelOperations`? Could you test the current implementation with subqueries inside the merge as well as matched/not matched conditions?

Reply:
@aokolnychyi will address this with a follow-up to not allow subqs and add tests to verify the same. Hope its okay.